src/previews/preview_service.rs

```diff
@@ -153,6 +153,33 @@ pub fn delete_file_preview(id: u32) {
     }
 }
 
+/// retrieves all file IDs from the database and publishes them to the preview generation queue.
+/// This function is intended to be called in a background thread as it may take some time
+/// to publish all messages depending on the number of files.
+pub fn load_all_files_in_preview_queue() {
+    use crate::repository::{file_repository, open_connection};
+    
+    let con = open_connection();
+    let file_ids = match file_repository::get_all_file_ids(&con) {
+        Ok(ids) => ids,
+        Err(e) => {
+            log::error!(
```

Comment:
Be sure to close the connection on all error cases before executing any other code, in order to avoid memory leaks